consensus/validator.go

```diff
@@ -172,17 +96,17 @@ func (consensus *Consensus) onPrepared(msg *msg_pb.Message) {
 		Uint64("MsgViewID", recvMsg.ViewID).
 		Msg("[OnPrepared] Received prepared message")
 
-	if recvMsg.BlockNum < consensus.blockNum {
+	if recvMsg.BlockNum != consensus.blockNum {
```

Comment:
We should still store the blocks with Num > curBlock, in the case of missed messages or messages out of order. But for old blocks for sure we can ignore.